Source/dthread.cpp

```diff
@@ -63,11 +64,10 @@ void DthreadHandler()
 
 void dthread_remove_player(uint8_t pnum)
 {
-	sgMemCrit.Enter();
+	std::lock_guard lock(DthreadMutex);

```

Comment:
CTAD is not available on some platforms (we're somewhere between C++11 and C++17). Use explicit template arguments to fix the build.